frontend/codegen.ts

```diff
@@ -0,0 +1,67 @@
+import { CodegenConfig, generate } from '@graphql-codegen/cli';
+
+const schemaConfig: CodegenConfig = {
+   overwrite: true,
+   generates: {
+      'lib/strapi-sdk/generated/schema.graphql': {
+         schema: 'http://localhost:1337/graphql',
+         plugins: ['schema-ast'],
+      },
+   },
+};
+
+const config: CodegenConfig = {
+   overwrite: true,
+   generates: {
+      'lib/strapi-sdk/generated/sdk.ts': {
+         schema: 'http://localhost:1337/graphql',
+         documents: 'lib/strapi-sdk/operations/**/*.graphql',
+         plugins: [
+            'typescript',
+            'typescript-operations',
+            'typescript-generic-sdk',
+         ],
+         hooks: {
+            afterOneFileWrite: ['prettier --write'],
+         },
+         config: {
+            documentMode: 'string',
+            dedupeFragments: true,
+         },
+      },
+      'lib/strapi-sdk/generated/validation.ts': {
+         schema: 'lib/strapi-sdk/generated/schema.graphql',
+         plugins: ['typescript-validation-schema'],
+         hooks: {
+            afterOneFileWrite: ['prettier --write'],
+         },
+         config: {
+            importFrom: './sdk',
+            schema: 'zod',
+         },
+      },
+      'lib/shopify-storefront-sdk/generated/sdk.ts': {
+         schema: 'lib/shopify-storefront-sdk/schema.graphql',
+         documents: 'lib/shopify-storefront-sdk/operations/**/*.graphql',
+         plugins: [
+            'typescript',
+            'typescript-operations',
+            'typescript-generic-sdk',
+         ],
+         hooks: {
+            afterOneFileWrite: ['prettier --write'],
+         },
+         config: {
+            documentMode: 'string',
+            dedupeFragments: true,
+            scalars: {
+               HTML: 'string',
+               URL: 'string',
+               Decimal: 'string',
+            },
+         },
+      },
+   },
+};
+
+export default generate(schemaConfig, true).then(() => config);
```

Comment:
Hmm... are there things that codegen does besides generate? i.e. now that the config file *always* generates, are there some cli commands (like `--help`) that used to just load the config and now they will end up running it (and generating files).

Perhaps this shouldn't be a config file, but instead a task: `generate(schemaConfig).then(() => generate(config))`

Reply:
I think codegen only evaluates `codegen.ts` when we pass the config file to the command as `--config` option. Like:

https://github.com/iFixit/react-commerce/blob/1c27e290a225e0980152f07d4a812858f1029848/frontend/package.json#L13

So we could run `pnpm graphql-codegen --help`, and the config file doesn't get evaluated.

That being said, this implementation _is_ a bit hacky. We're exporting a promise, though it seems to work despite that.

Reply:
My point was that if all we need is this file to be executed, we can have it call `generate()` twice and then run it directly (however that's done with `.ts` files) instead of having it get inadvertently "run" as part of being loaded as a config file.

Not mandatory, but it's rather weird to have a config file that ends up executing the program it's trying to configure as part of configuration.

Reply:
> Another option would be defining the two configurations separately, and running

I'd choose this option. I like that it's straight forward and the config doesn't execute the program it's intending to config.